app/storage/encrypted_questionnaire_storage.py

```diff
@@ -68,11 +68,39 @@ def _generate_key(user_id, user_ik, pepper):
         return to_bytes(sha256.hexdigest()[:32])
 
     def _encrypt_data(self, data):
-        encrypted = JWEDirEncrypter().encrypt(data, self._cek)
-        return {'data': encrypted}
+        protected_header = {
+            "alg": "dir",
+            "enc": "A256GCM",
+            "kid": "1,1",
+        }
+        token = jwe.JWE(plaintext=base64url_encode(data), protected=protected_header)
+
+        password = {
+            "kty": "oct",
+            "k": base64url_encode(self._cek),
+        }
+
+        key = jwk.JWK(**password)
+
+        token.add_recipient(key)
+
+        return {'data': token.serialize(compact=True)}
 
     def _decrypt_data(self, encrypted_data):
-        return JWEDirDecrypter().decrypt(encrypted_data['data'], self._cek)
+
+        encrypted_token = encrypted_data['data']
+
+        password = {
+            "kty": "oct",
+            "k": base64url_encode(self._cek),
+        }
+
+        key = jwk.JWK(**password)
```

Comment:
Might be worth factoring out creating the jwk into a method that takes the cek as a value. This method could be used by both the encrypt and decrypt methods.